pinc/ProjectTransition.inc

```diff
@@ -342,68 +344,49 @@ class ProjectTransition
 
 function project_pre_release_check($project, $round)
 // Check whether the project seems ready to be released into the round.
-// If it is, return an empty string.
-// If it has problems, return a detailed error-message.
+// If it is, return an empty array.
+// If it has problems, return an array of detailed error messages.
 {
-    global $projects_dir;
-
-    $errors = '';
+    if (!$project->pages_table_exists) {
+        return [_("There is no page-table.")];
+    }
```

Comment:
Once again, I'm not entirely sure whether I'm testing the right thing. I picked one of my projects (projectID5aa892e19f0bf) and renamed the project's page table to `projectID5aa892e19f0be`.

That immediately, when I refreshed the project page, threw a fatal error:

settings: state='P1.proj_unavail'

> Fatal error: Uncaught NoProjectPageTable: Project page table does not exist. in /home/cpeel/public_html/c.branch/update-proj-tran-queries/pinc/Project.inc:577 Stack trace: #0 /home/cpeel/public_html/c.branch/update-proj-tran-queries/pinc/Project.inc(597): Project->get_num_pages_in_state() `#1` /home/cpeel/public_html/c.branch/update-proj-tran-queries/pinc/project_edit.inc(58): Project->get_num_pages() `#2` /home/cpeel/public_html/c.branch/update-proj-tran-queries/project.php(899): user_can_add_project_pages() `#3` /home/cpeel/public_html/c.branch/update-proj-tran-queries/project.php(129): do_early_uploads() `#4` {main} thrown in /home/cpeel/public_html/c.branch/update-proj-tran-queries/pinc/Project.inc on line 577

Which made it kind of difficult to test out the button to transition to Waiting, since there weren't any buttons.

Then I remembered that I could maybe request the transition via the project search page. This is what I got for that:

> Fatal error: Uncaught DBQueryError: An error occurred during a database query and has been logged. in /home/cpeel/public_html/c.branch/update-proj-tran-queries/pinc/DPDatabase.inc:83 Stack trace: #0 /home/cpeel/public_html/c.branch/update-proj-tran-queries/pinc/DPage.inc(103): DPDatabase::query() `#1` /home/cpeel/public_html/c.branch/update-proj-tran-queries/pinc/ProjectTransition.inc(316): Pages_prepForRound() `#2` /home/cpeel/public_html/c.branch/update-proj-tran-queries/tools/changestate.php(116): ProjectTransition->do_state_change() `#3` {main} thrown in /home/cpeel/public_html/c.branch/update-proj-tran-queries/pinc/DPDatabase.inc on line 83

So, I'm not getting far enough to see this error message.

Reply:
Honestly, it wouldn't surprise me if you couldn't actually get here without a project table. I was just retaining the original "is it there" check.

Reply:
That makes sense, thanks.